examples/src/main/java/io/grpc/examples/advanced/HelloJsonServer.java

```diff
@@ -67,7 +68,11 @@ public void run() {
 
   private void stop() {
     if (server != null) {
-      server.shutdown();
+      try {
+        server.shutdown().awaitTermination(30, TimeUnit.SECONDS);
+      } catch (InterruptedException e) {
+        logger.warning(e.getMessage());
```

Comment:
So logging here is useless if called from a shutdown hook, because the logging system itself is shut down via a shutdown hook (and so this log goes nowhere).

It is probably best to do the awaitTermination()/blockUntilShutdown() from the shutdown hook and handle InterruptedException via `ex.printStackTrace()`. The other option would be to do awaitTermination() within stop() but have it throw InterruptedException, and then handle the InterruptedException within the hook via `ex.printStackTrace()`.

I really want to rewrite/reorganize the examples, since the start()/stop() methods just make things more complicated without a major benefit.

Reply:
Done in https://github.com/grpc/grpc-java/pull/6512/commits/8e75b2dd4ebe7db8fa0cb951d0868853c81b0350